module/zfs/vdev.c

```diff
@@ -2385,6 +2385,23 @@ vdev_copy_path_impl(vdev_t *svd, vdev_t *dvd)
 		zfs_dbgmsg("vdev_copy_path: vdev %llu: path set to '%s'",
 		    (u_longlong_t)dvd->vdev_guid, dvd->vdev_path);
 	}
+
+	/* Our enclosure sysfs path may have changed between imports */
+	if (svd->vdev_enc_sysfs_path != dvd->vdev_enc_sysfs_path) {
```

Comment:
Comparing pointers instead of comparing strings with strcmp()?

Reply:
Duh!  :man_facepalming:

Thank goodness for code reviews :smile:  I'll fix

Reply:
Thinking on this, is there ever a case when we want to keep the on-disk encpath? Wouldn't unconditionally using the encpath from userspace get the behavior we want?

Reply:
Correct, we always want to set it from userspace.  The comparison is only in there to print out the log message that the value changed.